### ISSUE TITLE
Attribute defaults return the same object

### ISSUE THREAD

**Issue by masayag** [fog/fog-core#250]
Having a Fog::Model that contains an attribute with default, i.e.[1]:
```
attribute :volumes,   :aliases => 'spec_volumes',   :default => []
```
When instantiating instances of the class, all of the attributes that have defaults get the same object as the default:
```
[125] pry(#<ForemanKubevirt::Kubevirt>)> Fog::Kubevirt::Compute::Server.new.networks.object_id
=> 127889820
[126] pry(#<ForemanKubevirt::Kubevirt>)> Fog::Kubevirt::Compute::Server.new.networks.object_id
=> 127889820
[127] pry(#<ForemanKubevirt::Kubevirt>)> Fog::Kubevirt::Compute::Server.new.networks.object_id
=> 127889820
```
So when updating one Server.networks attribute, all of the other server instances are being affected by that, since all reference to same object.

Is this a special trait of `:default` and expected behavior or a bug ?

[1] https://github.com/fog/fog-kubevirt/blob/master/lib/fog/kubevirt/compute/models/vm_base.rb#L20

**Comment by plribeiro3000** [comment]
Since the `attribute` method is a class definition we are storing all the DSL options on instance variables on the class itself so it is an expected behaviour.

Is this a problem for `fog-kubevirt`?

I was looking into the class you linked and it got me curious. Why are you defining a model as a Module?

**Comment by masayag** [comment]
> Since the `attribute` method is a class definition we are storing all the DSL options on instance variables on the class itself so it is an expected behaviour.
> 
> Is this a problem for `fog-kubevirt`?
> 

Yes, but I can simply overcome this by not using the defaults.
If we have a collection, i.e. networks for a Server model, I wouldn't expect all of the Server instances will be sharing the same networks array, rather each Server instance will have its own copy of the networks array.

Since this is by design, this issue can be closed.

> I was looking into the class you linked and it got me curious. Why are you defining a model as a Module?

We first developed the Fog::Model::Vm, as a class/model (that was used by manageiq), and when we developed the foreman_kubevirt plugin, we needed to implement the Server as well. 
On kubevirt Vm is a static definition of a Vm, where VmInstance is the the live object (similar to kubevirt api).
Server model is sort of a unification of both.
Declaring the VmBase as a module allowed us to share code between those two models (while VmBase is a Module, both Server and Vm are models that share great similarity).
